### DISCUSSION TITLE
Tutorial examples are not working

### DISCUSSION THREAD

**Discussion by dhananjayjha100** [sofa-framework/sofa#4100] [Write / Run a simulation] [answered]
I have just started using sofa, and I am using sofa v23.06 on windows and running this example oneTetrahedron, and these are the warnings/errors I am getting.
``` batch
[INFO]    [GUIManager] Registered batch as a GUI.
[INFO]    [GUIManager] Registered qglviewer as a GUI.
[INFO]    [GUIManager] Registered qt as a GUI.
[INFO]    [GUIManager] lastUsedGUI.ini not found; using default GUI.
[WARNING] [OglModel(visual)] Loading obj mesh file directly inside the VisualModel will be deprecated soon. Use a MeshOBJLoader and link the Data to the VisualModel. E.g:
<MeshOBJLoader name='myLoader' filename='myFilePath.obj'/>
<OglModel src='@myLoader'/>
[WARNING] [Node(root)] An animation loop is required, but has not been found. Add an animation loop in the root of your scene to fix this warning. The list of available animation loop components is: [ConstraintAnimationLoop, DefaultAnimationLoop, FreeMotionAnimationLoop, MultiStepAnimationLoop, MultiTagAnimationLoop]. A component of type DefaultAnimationLoop will be automatically added for you.
[WARNING] [CGLinearSolver(linearSolver)] Required data "iterations" has not been set. Falling back to default value: 25
[WARNING] [CGLinearSolver(linearSolver)] Required data "tolerance" has not been set. Falling back to default value: 1e-05
[WARNING] [CGLinearSolver(linearSolver)] Required data "threshold" has not been set. Falling back to default value: 1e-05
[INFO]    [SofaPluginManager] Loading automatically plugin list in ./loadedPlugins.ini
[INFO]    [PluginManager] 0 plugins have been loaded from ./loadedPlugins.ini

C:\libSofa\sofabuild\bin\Release\oneTetrahedron.exe (process 14876) exited with code -1073741819.
Press any key to close this window . . .
```

Can anyone help me how can I proceed to fix this,
thanks

**Comment by dusevitch** [comment]
Is there a specific reason you're using C++ as opposed to running Python? For me starting out, running the Python code and getting that to work with sofa was a lot easier to learn the software.

**Reply by dhananjayjha100** [reply]
Yes, actually, I need to resume an old sofa simulation project that was written in C++. My task involves reviving that code, so I want to focus more on C++ examples to gain a better understanding.

**Comment by alxbilger** [comment]
This tutorial has not been updated for a long time. There are chances that it no longer works. The warnings are not critical. We'll have a look and let you know.

**Reply by alxbilger** [reply, marked as answer]
@dhananjayjha100 I made two pull requests that should fix the crash: #4103 and #4102.

**Reply by dhananjayjha100** [reply]
Thankyou @alxbilger for your support. It fixed all the warnings but the compiler throws error at the following line:

Error is:
`namespace "sofa::simulation::node" has no member "initRoot"`
The namespace "sofa::simulation::node" is in Sofa/framework/Simulation/Core/src/sofa/simulation/fwd.cpp as:
```
namespace sofa::simulation::node
{
sofa::simulation::Node* getNodeFrom(sofa::core::objectmodel::BaseContext* context)
{
    return dynamic_cast<sofa::simulation::Node*>(context);
}

sofa::core::objectmodel::BaseContext* toBaseContext(sofa::simulation::Node* node)
{
    return static_cast<sofa::core::objectmodel::BaseContext*>(node);
}
}
```

Also I found that the simulation crashes during execution of this line:
`const auto groot = sofa::simulation::getSimulation()->createNewGraph("root");`

So, is there something I can do to fix this.
Thanks

**Reply by alxbilger** [reply]
This is because the new code is based on the current master branch of Sofa. What version do you have?

**Reply by dhananjayjha100** [reply]
I have the v23.06 stable version

**Reply by dhananjayjha100** [reply]
Thankyou so much @alxbilger, I did the changes on the new code, and it worked.

**Reply by alxbilger** [reply]
Great!
